OpenRA.Game/Network/UnitOrders.cs

```diff
@@ -67,6 +67,35 @@ public static void ProcessOrder(OrderManager orderManager, World world, int clie
 						break;
 					}
 
+				case "ReplacePlayerWithBot":
+					{
+						var client = orderManager.LobbyInfo.ClientWithIndex(clientId);
+						if (client != null && world != null)
+						{
+							var player = world.FindPlayerByClient(client);
+							if (player == null || player.IsBot)
+								break;
+
+							client.State = Session.ClientState.Ready;
+
+							var admin = orderManager.LobbyInfo.Clients.FirstOrDefault(c => c.IsAdmin);
```

Comment:
Is there any particular reason the bot controller has to be admin ?
If not couldn't you select a play at random, assign the bots to them and if the dropped player had any bots remake them for different random players ?